### PR TITLE
Add initial implementation for the package

### LINKED ISSUE #7240: Introduce a Ballerina connector for IBM CICS Transaction Gateway
**Description:**
> $subject

Related to: [#207](https://github.com/wso2-enterprise/integration-product-management/issues/207)

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes: [#7240](https://github.com/ballerina-platform/ballerina-library/issues/7240)

## Examples
N/A

## Checklist
- [X] Linked to an issue
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,100 @@
...
+# Ballerina IBM CICS Transaction Gateway connector
+
+[![Build](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/actions/workflows/ci.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/actions/workflows/ci.yml)
+[![Trivy](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/actions/workflows/trivy-scan.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/actions/workflows/trivy-scan.yml)
+[![GraalVM Check](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/actions/workflows/build-with-bal-test-graalvm.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/actions/workflows/build-with-bal-test-graalvm.yml)
+[![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerinax-ibm.ctg.svg)](https://github.com/ballerina-platform/module-ballerinax-ibm.ctg/commits/main)
+[![GitHub Issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-library/module/ibm.ctg.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-library/labels/module%2Fibm.ctg)
+
```
Comment (NipunaRanasinghe) [inline README.md:8]: Are we planning to add setup guide, quickstart and examples in a separate PR?

File: ballerina/types.bal [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    int port;
+    # CICS server name
+    string cicsServer;
+    # The authentication configurations for the CICS server
+    Auth auth;
+    # The SSL configurations for the CICS transaction gateway
+    SecureSocket secureSocket?;
+|};
+
+# Represents the client authentication configurations for CICS server.
```
Comment (NipunaRanasinghe) [inline ballerina/types.bal]:
```suggestion
# Represents the client authentication configuration for CICS server.
```

File: ballerina/types.bal [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+public type Auth record {|
+    # The CICS userId
+    string userId;
+    # The CICS password
+    string password;
+|};
+
+# Represents the SSL configurations for CICS transaction gateway.
```
Comment (NipunaRanasinghe) [inline ballerina/types.bal]:
```suggestion
# Represents the SSL configuration for CICS transaction gateway.
```

**Review by NipunaRanasinghe — COMMENTED** [review]
@ayeshLK there seems to be a build failure.Please check

File: README.md (hunk 1 above)
Comment (ayeshLK) [reply README.md:8]: Yes, that is the plan.

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**